src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/split-pane-header/split-pane-header.component.ts

```
@@ -14,16 +16,87 @@
+        this.fileSelectSubscription = this.fileSelectedSubject.subscribe((val) => {
+            if (this.isLockFilesEnabled) {
+                this.handleLockedFileSelection(val.file, val.idx);
+            }
+        });
+    }
+
+    private handleLockedFileSelection(file: FileWithHasMatch, idx: number): void {
+        let index;
+        if (this.files[idx]?.file === file.file) {
+            this.handleFileSelectWithoutPropagation(file, idx);
+        } else if ((index = this.getIndexOf(file)) > 0) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)